compiler/executable/match_/planner/vertex/constraint.rs

```diff
@@ -124,32 +124,34 @@ impl ConstraintVertex<'_> {
 
     pub(crate) fn direction_from_join_var(
         &self,
-        var: VariableVertexId,
-        include: &HashSet<VariableVertexId>,
-        exclude: &HashSet<VariableVertexId>,
-    ) -> Option<Direction> {
+        join_var: VariableVertexId,
+        _ongoing_step_produced: &HashSet<VariableVertexId>,
```

Comment:
This is clearly used.
```suggestion
        ongoing_step_produced: &HashSet<VariableVertexId>,
```